sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -34,70 +33,69 @@ class KlaviyoPushService : FirebaseMessagingService() {
          * @param pushToken The push token provided by the FCM Service
          */
         fun setPushToken(pushToken: String) {
-            val properties = KlaviyoCustomerProperties()
-                .addAppendProperty(REQUEST_PUSH_KEY, pushToken)
+            val profile = Profile().addAppendProperty(PUSH_TOKEN_APPEND_KEY, pushToken)
 
-            Klaviyo.identify(properties)
+            Klaviyo.setProfile(profile)
 
-            KlaviyoPreferenceUtils.writeStringPreference(PUSH_TOKEN_PREFERENCE_KEY, pushToken)
+            Registry.dataStore.store(PUSH_TOKEN_KEY, pushToken)
         }
 
         /**
-         * Returns the current push token that we have stored on this device
+         * Retrieves the device FCM push token stored on this device
          *
          * @return The push token we read from the shared preferences
          */
-        fun getCurrentPushToken(): String {
-            return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
+        internal fun getPushToken(): String {
+            return Registry.dataStore.fetch(PUSH_TOKEN_KEY) ?: ""
```

Comment:
should we handle null more explicitly downstream?

Reply:
looks like we only using it in `openedPush` to include push tokens, to slim the payload, shouldnt we no include the `PUSH_TOKEN_KEY` if there is no token?

Reply:
True, and I think this will go back to being a public function again too, and returning null is a clearer empty state than just empty string.